Wrappers/Python/ccpi/viewer/utils/io.py

```diff
@@ -43,3 +71,761 @@ def SaveRenderToPNG(render_window, filename):
+                    # VTK's order is Fortran whereas h5py writes in
+                    # C order. We don't want to do deep copies so we write
+                    # with shape flipped and pretend the array is
+                    # C order.
+                    array = array.reshape(data.GetDimensions()[::-1])
+                else:
+                    array = None
+                try:
+                    if array is None:
+                        dset = f.create_dataset(dataset_name, dataset_info['shape'])
+                    else:
+                        if self._Chunking:
+                            if self._ChunkShape is None:
+                                # If Chunking has been selected but a shape has not,
+                                # by default use a slice as the chunk
+                                slice_shape = list(array.shape)
+                                slice_shape[0] = 1
+                                self._ChunkShape = tuple(slice_shape)
+                            if self._HDF5Compression is None or self._HDF5Compression[0] is None:
+                                dset = f.create_dataset(dataset_name, data=array,
+                                                chunks=self._ChunkShape)                  
+                            elif self._HDF5Compression[0] == 'gzip':
+                                dset = f.create_dataset(dataset_name, data=array, 
+                                                    chunks=self._ChunkShape, 
+                                                    compression=self._HDF5Compression[0], 
+                                                    compression_opts=self._HDF5Compression[1], 
+                                                    shuffle=self._HDF5Compression[2]
+                                                    )                  
+                            elif self._HDF5Compression[0] == 'lzf':
+                                dset = f.create_dataset(dataset_name, data=array, 
+                                                    chunks=self._ChunkShape, 
+                                                    compression=self._HDF5Compression[0], 
+                                                    shuffle=self._HDF5Compression[2]
+                                                    )
+                        else:
+                            dset = f.create_dataset(dataset_name, data=array)
+
+                except RuntimeError:
+                        print("Unable to save image data to {0}."
+                            "Dataset with name {1} already exists in this file.".format(
+                                self._FileName, dataset_name))
+
+                if entry_num != 1:
+                    dset.attrs['original_dataset'] = 'entry1/tomo_entry/data/data'
+
+                for key, value in dataset_info.items():
+                    dset.attrs[key] = value
+
+
+class vortexHDF5ImageReader(HDF5Reader):
```

Comment:
should we name this something else? Are we committed to renaming the module to vortex?

Reply:
I don't know if we need `vortex` in the class name. I think it could be removed. It will be found in the `vortex` package anyway.

Is the path to the data `entry{}/tomo_entry/data/data` not part of the NeXuS specification?
The class name is now `HDF5ImageReader`, what is it meant by `Image`?

Reply:
well the thing is we already have a HDF5 reader which is called HDF5Reader, so it shouldn't have the same name as that. This class is for a HDF5 type that we have defined that we expect to include certain info - the origin and spacing.
Kind of the viewer's equivalent of the CIL Nexus reader expecting certain attributes etc.

Reply:
yes you are right, `entry{}/tomo_entry/data/data` is part of the NeXuS specification.
We do not have a specific NeXuS reader in the viewer yet just a generic HDF5 reader.
We should have a NeXuS one, and then this could inherit from the NeXuS one. I can either do that now or in another PR.

Reply:
What do we rename to now that we keep cilviewer as the name? Maybe cilviewerHDF5Reader?

Reply:
renamed to cilviewerHDF5Reader